static/sass/initial/_type.scss

```diff
@@ -57,30 +57,43 @@ h3 {
 a {
   #{$link} {
     @include text-underline('action');
-    color: inherit;
+    color: color('action');
     font-weight: bold;
     text-decoration: none;
     transition: color 150ms;
+
+    h1 &,
+    h2 &,
+    h3 & {
+      background-image: none; // remove link underline for headlines
+    }
   }
 
   #{$focus} {
-    color: color('action');
+    background-image: linear-gradient(color('action-hover'), color('action-hover'));
+    color: color('action-hover');
   }
 }
 
+/// @todo totally yucky, need a better class to hook into
+///   [data-module='rst-body'] is only for blog posts
+///   and no global inner-level class exists
```

Comment:
Have you played with `config.pageclasses` at all? You can add `pageclasses: longform` in your YAML content at the top of a page/post, and it will get added as a class on `data-region="main"`. (That can all be moved around as you like, and it's easy to add new config like that. maybe `longform: yes | no`?